--- ./dub/build.d
+++ ./dub/build.d
@@ -84,7 +84,7 @@
         ] ~ dflags ~ [ "@build-files.txt" ];
 
     writeln("Building dub using ", dmd, " (dflags: ", dflags, "), this may take a while...");
-    auto proc = execute(command);
+    auto proc = execute(command, null, Config.inheritFDs | Config.stderrPassThrough);
     if (proc.status != 0)
     {
         writeln("Command `", command, "` failed, output was:");
