Alchemy patch for LDC v1.13.0 to v1.21.0-beta1.
--- ./ldc-source/driver/linker-gcc.cpp
+++ ./ldc-source/driver/linker-gcc.cpp
@@ -524,13 +524,6 @@
 void ArgsBuilder::addLinker() {
   if (!opts::linker.empty()) {
     args.push_back("-fuse-ld=" + opts::linker);
-  } else if (global.params.isLinux &&
-             global.params.targetTriple->getEnvironment() !=
-                 llvm::Triple::Android) {
-    // Default to ld.gold on Linux due to ld.bfd issues with ThinLTO (see #2278)
-    // and older bfd versions stripping llvm.used symbols (e.g., ModuleInfo
-    // refs) with --gc-sections (see #2870).
-    args.push_back("-fuse-ld=gold");
   }
 }
 
