Alchemy patch for LDC v1.21.0 to v1.22.0-beta2.
--- ./ldc-source/driver/linker-gcc.cpp
+++ ./ldc-source/driver/linker-gcc.cpp
@@ -571,15 +571,6 @@
 void ArgsBuilder::addLinker() {
   if (!opts::linker.empty()) {
     args.push_back("-fuse-ld=" + opts::linker);
-  } else if (global.params.isLinux &&
-             global.params.targetTriple->getEnvironment() !=
-                 llvm::Triple::Android) {
-    // Default to ld.gold on Linux due to ld.bfd issues with ThinLTO (see #2278)
-    // and older bfd versions stripping llvm.used symbols (e.g., ModuleInfo
-    // refs) with --gc-sections (see #2870).
-    // Can be overridden by `-linker=` (explicitly empty).
-    if (opts::linker.getNumOccurrences() == 0)
-      args.push_back("-fuse-ld=gold");
   }
 }
 
