Alchemy patch for LDC v1.22.0 to v1.24.0.
--- ./ldc-source/driver/linker-gcc.cpp
+++ ./ldc-source/driver/linker-gcc.cpp
@@ -571,23 +571,6 @@
 void ArgsBuilder::addLinker() {
   llvm::StringRef linker = opts::linker;
 
-  // We have a default linker preference for Linux targets. It can be disabled
-  // via `-linker=` (explicitly empty).
-  if (global.params.isLinux && opts::linker.getNumOccurrences() == 0) {
-    // Default to ld.bfd for Android (placing .tdata and .tbss sections adjacent
-    // to each other as required by druntime's rt.sections_android, contrary to
-    // gold and lld as of Android NDK r21d).
-    if (global.params.targetTriple->getEnvironment() == llvm::Triple::Android) {
-      linker = "bfd";
-    }
-    // Otherwise default to ld.gold for Linux due to ld.bfd issues with ThinLTO
-    // (see #2278) and older bfd versions stripping llvm.used symbols (e.g.,
-    // ModuleInfo refs) with --gc-sections (see #2870).
-    else {
-      linker = "gold";
-    }
-  }
-
   if (!linker.empty())
     args.push_back(("-fuse-ld=" + linker).str());
 }
